✨(react) make optional DataGrid row selection (#339)
openfun/cunningham: packages/react/src/components/DataGrid/index.mdx

<Canvas sourceState="shown" of={Stories.ClientSideWithPagination}/>

> You can also make a row non selectable by setting using `enableRowSelection` as a callback function that returns a boolean based on a condition.

Reviewer: ```suggestion
You can also make a row non selectable by setting `enableRowSelection` prop with a callback function that returns a boolean based on a condition.
```
